charts/prefect-server/templates/deployment.yaml

```diff
@@ -82,6 +82,10 @@ spec:
             - name: PREFECT_UI_API_URL
               value: {{ .Values.server.publicApiUrl | quote }}
             {{- end }}
+            {{- if .Values.server.apiHost }}
+            - name: PREFECT_SERVER_API_HOST
```

Comment:
Why is this setting necessary, can you explain?

The public API URL (the URL that the frontend uses to communicate with the backend) should be sufficient for most use cases.

We listen to all interfaces (albeit, currently, all IPv4 interfaces only) and that's set here: https://github.com/PrefectHQ/prefect-helm/blob/b694a97f614a52fdfa58cc8eca6faa59ebc22e77/charts/prefect-server/templates/deployment.yaml#L64-L68

You should be able to use a LoadBalancer service to route traffic to the OSS Prefect server without setting anything else

Reply:
It was a belt-and-suspenders thing. Based on https://github.com/PrefectHQ/prefect/blob/main/RELEASE-NOTES.md#20b5, since the API_HOST can be used to build the URL for API access, I wanted to make sure all possible paths for API URL building could be set to the proper host.